### DISCUSSION TITLE
Unexpected behaviour of Unique Constraints bundle

### DISCUSSION THREAD

**Discussion by oyvindvalland** [ravendb/ravendb#12889] [Q&A] [answered]
Hello,

This question relates to a Raven 3.5.9 database.

We use the Unique Constraints bundle for a few things and it works very well. However, I have just discovered something that surprises me and it has caused a problem in a development database.

We have a document where we use the Unique Constraints on a property of type `long`.  This results in unique constraints documents with ids on the form `uniqueconstraints/thedocumenttype/thepropertyname/1234`  as you would expect.

However, I've found that sometimes we also get a second unique constraints document where the `long` value appears to be treated as a decimal: `uniqueconstraints/thedocumenttype/thepropertyname/1234.0`

That's not a problem so long as the first document exists. But now we've had a scenario where only the latter document was created, and we cannot do a lookup by unique constraint because `1234 != 1234.0`

I should add that this particular case happened when we inserted a document into the db via `DatabaseCommands.Batch` and a `PutCommandData` command.

Documents created with the RavenDb client appear to get either the correct unique constraints document, or _both_ unique constraints documents.

Any idea what's going on here? Is there a way to fix or work around this?

**Comment by ayende** [comment, marked as answer]
* Note that 3.5 is no longer support.
* The issue is likely related to how you wrote the data, when writing via the C# API, we can tell what the type is, when you are working with the studio, all numbers are actually double, so that is what you are seeing here.

**Reply by oyvindvalland** [reply]
Thank you. That does make sense  - we can work around it for some scenarios where required.
